spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/FloController.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package org.springframework.xd.dirt.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.VndErrors;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.xd.dirt.stream.MultiStreamParseResult;
+import org.springframework.xd.dirt.stream.XDStreamParser;
+import org.springframework.xd.module.ModuleDescriptor;
+
+/**
+ * A controller that can accept a whole list of definitions and report whether
+ * they are valid as a whole or not.
+ *
+ * @author Eric Bottard
+ */
+@RestController
+@RequestMapping("/flo")
+public class FloController {
```

Comment:
Similar comment to below, I don't really want flo specifics in XD. I want generally reusable endpoints that others may also want to use. I think a general parse endpoint makes a lot of sense. What seems to make things trickier here is that the syntax processing is mixed up a bit with the semantic processing in the parser. So if I want to parse two related streams, I have to use the TransientDefinitionRepository. If the syntax parsing was entirely separate from any semantic concerns (as I know we'd all like to do), I could simply call a parse endpoint that took a single DSL string. And it wouldn't matter if it referred to another stream that didn't exist yet. But because of those checks in the parser I have to pass in multiple strings so the verification checks can pass.  Is it worth trying to do that work now I wonder: pull out the parser semantic comprehension (may not be a lot of it), then you can just give me a single DSL string parsing endpoint.